### PR TITLE
ci: build for e2e

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Opening the shadow root mode for the chromium-e2e and firefox-e2e builds.

Closes https://mask.atlassian.net/browse/MF-540

image

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

**Comment by Jack-Works** [comment]
do we have plan to introduce e2e again?

**Comment by guanbinrui** [comment]
> do we have plan to introduce e2e again?

Yep, but originate by the @DimensionDev/qa team.

**Review by Jack-Works — APPROVED** [review]

File: packages/mask/.webpack/config.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export function createConfiguration(_inputFlags: BuildFlags): Configuration {
             })(),
             new EnvironmentPlugin({
                 NODE_ENV: flags.mode,
-                shadowRootMode: flags.devtools ? 'open' : 'closed',
+                shadowRootMode: flags.devtools || flags.channel === 'e2e' ? 'open' : 'closed',
```
Comment (Jack-Works) [inline packages/mask/.webpack/config.ts:178]: what if we let beta&nightly versions have opened shadow root?

Comment (Jack-Works) [reply packages/mask/.webpack/config.ts:178]: then we can avoid two more channels

Comment (guanbinrui) [reply packages/mask/.webpack/config.ts:178]: Done here: https://github.com/DimensionDev/Maskbook/pull/8130

**Pull request closed without merging**